pkg/okteto/pipeline.go

```diff
@@ -104,91 +117,83 @@ type deprecatedDestroyPipelineResponse struct {
-			variablesArg = append(variablesArg, InputVariable{
-				Name:  graphql.String("OKTETO_ORIGIN"),
-				Value: graphql.String(origin),
-			})
-		}
-		queryVariables := map[string]interface{}{
-			"name":       graphql.String(opts.Name),
-			"repository": graphql.String(opts.Repository),
-			"space":      graphql.String(opts.Namespace),
-			"branch":     graphql.String(opts.Branch),
-			"variables":  variablesArg,
-			"filename":   graphql.String(opts.Filename),
-		}
 
-		err := mutate(ctx, &mutation, queryVariables, c.client)
+	mutationVariables := c.getDeployVariables(opts)
+	var response deployPipelineResponse
+	if len(opts.Labels) == 0 {
+		mutationStruct := &deployPipelineMutation{}
+		err := mutate(ctx, mutationStruct, mutationVariables, c.client)
 		if err != nil {
 			return nil, fmt.Errorf("failed to deploy pipeline: %w", err)
 		}
-
-		gitDeployResponse.Action = &types.Action{
-			ID:     string(mutation.Response.Action.Id),
-			Name:   string(mutation.Response.Action.Name),
-			Status: string(mutation.Response.Action.Status),
-		}
-		gitDeployResponse.GitDeploy = &types.GitDeploy{
-			ID:         string(mutation.Response.GitDeploy.Id),
-			Name:       string(mutation.Response.GitDeploy.Name),
-			Repository: string(mutation.Response.GitDeploy.Repository),
-			Status:     string(mutation.Response.GitDeploy.Status),
-		}
-
+		response = mutationStruct.Response
 	} else {
-		var mutation deployPipelineMutation
-		queryVariables := map[string]interface{}{
-			"name":       graphql.String(opts.Name),
-			"repository": graphql.String(opts.Repository),
-			"space":      graphql.String(opts.Namespace),
-			"branch":     graphql.String(opts.Branch),
-			"filename":   graphql.String(opts.Filename),
-			"variables": []InputVariable{
-				{Name: graphql.String("OKTETO_ORIGIN"), Value: graphql.String(origin)},
-			},
-		}
+		mutationStruct := &deployPipelineMutationWithLabels{}
+		err := mutate(ctx, mutationStruct, mutationVariables, c.client)
 
-		err := mutate(ctx, &mutation, queryVariables, c.client)
 		if err != nil {
+			if strings.Contains(err.Error(), "Unknown argument \"labels\" on field \"deployGitRepository\" of type \"Mutation\"") {
+				return nil, oktetoErrors.UserError{E: ErrDeployPipelineLabelsFeatureNotSupported, Hint: "Please upgrade to the latest version or ask your administrator"}
+			}
 			return nil, fmt.Errorf("failed to deploy pipeline: %w", err)
 		}
+		response = mutationStruct.Response
+	}
 
-		gitDeployResponse.Action = &types.Action{
-			ID:     string(mutation.Response.Action.Id),
-			Name:   string(mutation.Response.Action.Name),
-			Status: string(mutation.Response.Action.Status),
-		}
-		gitDeployResponse.GitDeploy = &types.GitDeploy{
-			ID:         string(mutation.Response.GitDeploy.Id),
-			Name:       string(mutation.Response.GitDeploy.Name),
-			Repository: string(mutation.Response.GitDeploy.Repository),
-			Status:     string(mutation.Response.GitDeploy.Status),
+	gitDeployResponse := &types.GitDeployResponse{
```

Comment:
aren't we returning the labels as field?

Reply:
no, We just return the action ID and git deploy which are the necessary fields in order to show logs and know if it's been deployed.